pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/HLRealtimeSegmentDataManager.java

```diff
@@ -121,15 +123,18 @@ public HLRealtimeSegmentDataManager(final RealtimeSegmentZKMetadata segmentMetad
         this.sortedColumn = null;
       }
     }
-    //inverted index columns
-    invertedIndexColumns = indexingConfig.getInvertedIndexColumns();
-    if (sortedColumn != null && !invertedIndexColumns.contains(sortedColumn)) {
+
+    // Inverted index columns
+    Set<String> invertedIndexColumns = indexLoadingConfig.getInvertedIndexColumns();
+    if (sortedColumn != null) {
```

Comment:
I know this is not part of your PR, but not sure if I understand the motivation of adding inv index when sortedColumn != null. Could you add an item to investigate this further (outside of this PR)?